nfstream/plugins/ja4.py

```diff
@@ -0,0 +1,666 @@
+from nfstream import NFPlugin
+from scapy.all import IP, TCP, UDP
+import hashlib
+
+# JA4 construction functions
+def get_protocol(packet):
+    """Protocol detection function, also returns packet payload (TCP or UDP)
+        (protocol detection will be used in the future, but for the moment
+        it's not necessary, since only TCP fingerprints are generated)
+    Args:
+        packet (NFPacket): selected packet
+
+    Returns:
+        protocol (str): detected protocol, "t" for TCP, "q" for QUIC
+        payload (scapy.layers.inet.TCP or scapy.layers.inet.UDP): payload of the packet (at transport layer level)
+    """
+    # TCP detection and TCP payload extraction
+    if IP(packet.ip_packet).haslayer(TCP):
+        assert packet.protocol == 6
+        return ("t", IP(packet.ip_packet)[TCP])
+    # QUIC detection and QUIC payload extraction
+    elif IP(packet.ip_packet).haslayer(UDP):
+        assert packet.protocol == 17
```

Comment:
What is the purpose of using `assert` within this plugin?

Reply:
Now that I think about it... None. Already deleted and commited :-)